opppy/tally.py

```diff
@@ -239,25 +246,58 @@ def append_tally_dictionary(data, output_files, opppy_parser, append_date=False)
     total = len(output_files) 
     print('')
     print("Number of files to be read: ", total)
-    cycle_string_list=[]
-    for file_name in output_files:
-      cycle_string_list+=get_output_lines(file_name, opppy_parser.cycle_opening_string, opppy_parser.cycle_closing_string, opppy_parser.file_end_string)
-      if 'appended_files' in data:
-          data['appended_files'].append(file_name.split('/')[-1]+time)
-      else:
-          data['appended_files'] = [file_name.split('/')[-1]+time]
-      count += 1
-      progress(count,total, 'of input files read')
+    data_list = []
+    if(USE_THREADS):
+      def thread_all(file_name, result_d):
+          thread_cycle_string_list = get_output_lines(file_name, opppy_parser.cycle_opening_string, opppy_parser.cycle_closing_string, opppy_parser.file_end_string)
+          thread_data=[]
+          for cycle_string in thread_cycle_string_list:
+              print(cycle_string)
+              thread_data.append(extract_cycle_data(cycle_string, opppy_parser))
+          print(thread_data)
+          result_d[file_name] = thread_data
 
-    total = len(cycle_string_list) 
-    count = 0
-    print('')
-    print("Number of cycles to be parsed: ", total)
-    for cycle_string in cycle_string_list:
-      cycle_data = extract_cycle_data(cycle_string, opppy_parser)
-      data = append_tally_data(cycle_data,data,opppy_parser.sort_key_string)
-      count += 1
-      progress(count,total, 'of cycles parsed')
+      with Manager() as manager:
+            result_d = manager.dict()
+            threads = []
+            for file_name in output_files:
+                thread = Process(target=thread_all, args=(file_name, result_d,))
+                thread.start()
+                threads.append(thread)
+            for thread in threads:
+                thread.join()
+                count += 1
+                progress(count,total, 'of input files read')
+            for file_name in output_files:
+                data_list += result_d[file_name]
+    else:
```

Comment:
Pretty much the same comment here, you're now covering the parallel code with tests, but not the serial code, and the testing in `test_append_tally_dictionary` doesn't appear to by very robust--it just prints stuff without asserting on properties, so that makes me a little nervous.

Reply:
This is a fair point about not verifying properties. I'll have to think about the best way to check this data. Mostly this has been done with visualization.

Reply:
I've added gold files and value checks to test_interactive, test_dump_utils, test_outputs, and test_tallies. This should make the testing significantly more robust. I also added a test for the append_dump function which was previously untested.